fix(components): [date-picker] default-time property not working when selecting dates multiple times

SpanManX opened the pull request:
**多次选择日期 default-time 属性失效**

使用 `default-time` 设置默认时间，第一次选择日期时会正确返回 `default-time` 设置的时间，第二次选择日期时返回的是 `00:00:00`。`default-time` 属性在 `type="datetimerange"` 确是正常的，多次选择日期都会返回 default-time 设置的时间

>**Default-time property not working when selecting dates multiple times**
When using `default-time` to set the default time, the first date selection correctly returns the set time. However, on the second selection, it returns `00:00:00`. The `default-time` property works correctly when `type="datetimerange"`, consistently returning the set time for multiple selections.

[demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBkZWZhdWx0VGltZTEgPSBuZXcgRGF0ZSgyMDAwLCAxLCAxLCAxMiwgMCwgMCkgLy8gJzEyOjAwOjAwJ1xuY29uc3QgdmFsdWUxID0gcmVmKCcnKVxuY29uc3QgdmFsdWUyID0gcmVmKCcnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGgxPldyb25nPC9oMT5cbiAgPHA+e3sgdmFsdWUxIH19PC9wPlxuICAgICAgPGVsLWRhdGUtcGlja2VyXG4gICAgICAgIHYtbW9kZWw9XCJ2YWx1ZTFcIlxuICAgICAgICB0eXBlPVwiZGF0ZVwiXG4gICAgICAgIHBsYWNlaG9sZGVyPVwiUGljayBhIGRheVwiXG4gICAgICAgIDpkZWZhdWx0LXRpbWU9XCJuZXcgRGF0ZSgyMDAwLCAxLCAxLCAyMywgNTksIDU5KVwiXG4gICAgICAgICB2YWx1ZS1mb3JtYXQ9XCJZWVlZLU1NLUREIEhIOm1tOnNzXCJcbiAgICAgIC8+XG5cbiAgICAgIDxoMT5SaWdodDwvaDE+XG4gIDxwPnt7IHZhbHVlMiB9fTwvcD5cbiAgICA8ZWwtZGF0ZS1waWNrZXJcbiAgICAgIHYtbW9kZWw9XCJ2YWx1ZTJcIlxuICAgICAgdHlwZT1cImRhdGV0aW1lcmFuZ2VcIlxuICAgICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCBEYXRlXCJcbiAgICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCBEYXRlXCJcbiAgICAgIDpkZWZhdWx0LXRpbWU9XCJkZWZhdWx0VGltZTFcIlxuICAgICAgdmFsdWUtZm9ybWF0PVwiWVlZWS1NTS1ERCBISDptbTpzc1wiXG4gICAgLz5cbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgZXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vZWxlbWVudC1wbHVzQGxhdGVzdC9kaXN0L2luZGV4LmNzcycsICdodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL2VsZW1lbnQtcGx1c0BsYXRlc3QvdGhlbWUtY2hhbGsvZGFyay9jc3MtdmFycy5jc3MnXS5tYXAoKHN0eWxlKSA9PiB7XG4gICAgcmV0dXJuIG5ldyBQcm9taXNlKChyZXNvbHZlLCByZWplY3QpID0+IHtcbiAgICAgIGNvbnN0IGxpbmsgPSBkb2N1bWVudC5jcmVhdGVFbGVtZW50KCdsaW5rJylcbiAgICAgIGxpbmsucmVsID0gJ3N0eWxlc2hlZXQnXG4gICAgICBsaW5rLmhyZWYgPSBzdHlsZVxuICAgICAgbGluay5hZGRFdmVudExpc3RlbmVyKCdsb2FkJywgcmVzb2x2ZSlcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignZXJyb3InLCByZWplY3QpXG4gICAgICBkb2N1bWVudC5ib2R5LmFwcGVuZChsaW5rKVxuICAgIH0pXG4gIH0pXG4gIHJldHVybiBQcm9taXNlLmFsbFNldHRsZWQoc3R5bGVzKVxufVxuIiwidHNjb25maWcuanNvbiI6IntcbiAgXCJjb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IFwiRVNOZXh0XCIsXG4gICAgXCJqc3hcIjogXCJwcmVzZXJ2ZVwiLFxuICAgIFwibW9kdWxlXCI6IFwiRVNOZXh0XCIsXG4gICAgXCJtb2R1bGVSZXNvbHV0aW9uXCI6IFwiQnVuZGxlclwiLFxuICAgIFwidHlwZXNcIjogW1wiZWxlbWVudC1wbHVzL2dsb2JhbC5kLnRzXCJdLFxuICAgIFwiYWxsb3dJbXBvcnRpbmdUc0V4dGVuc2lvbnNcIjogdHJ1ZSxcbiAgICBcImFsbG93SnNcIjogdHJ1ZSxcbiAgICBcImNoZWNrSnNcIjogdHJ1ZVxuICB9LFxuICBcInZ1ZUNvbXBpbGVyT3B0aW9uc1wiOiB7XG4gICAgXCJ0YXJnZXRcIjogMy4zXG4gIH1cbn1cbiIsIlBsYXlncm91bmRNYWluLnZ1ZSI6IjxzY3JpcHQgc2V0dXA+XG5pbXBvcnQgQXBwIGZyb20gJy4vQXBwLnZ1ZSdcbmltcG9ydCB7IHNldHVwRWxlbWVudFBsdXMgfSBmcm9tICcuL2VsZW1lbnQtcGx1cy5qcydcbnNldHVwRWxlbWVudFBsdXMoKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPEFwcCAvPlxuPC90ZW1wbGF0ZT5cbiIsImltcG9ydC1tYXAuanNvbiI6IntcbiAgXCJpbXBvcnRzXCI6IHtcbiAgICBcInZ1ZVwiOiBcImh0dHBzOi8vY2RuLmpzZGVsaXZyLm5ldC9ucG0vQHZ1ZS9ydW50aW1lLWRvbUBsYXRlc3QvZGlzdC9ydW50aW1lLWRvbS5lc20tYnJvd3Nlci5qc1wiLFxuICAgIFwiQHZ1ZS9zaGFyZWRcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvc2hhcmVkQGxhdGVzdC9kaXN0L3NoYXJlZC5lc20tYnVuZGxlci5qc1wiLFxuICAgIFwiZWxlbWVudC1wbHVzXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwiaHR0cHM6Ly9jZG4uanNkZWxpdnIubmV0L25wbS9lbGVtZW50LXBsdXNAbGF0ZXN0L1wiLFxuICAgIFwiQGVsZW1lbnQtcGx1cy9pY29ucy12dWVcIjogXCJodHRwczovL2Nkbi5qc2RlbGl2ci5uZXQvbnBtL0BlbGVtZW50LXBsdXMvaWNvbnMtdnVlQDIvZGlzdC9pbmRleC5taW4uanNcIlxuICB9LFxuICBcInNjb3Blc1wiOiB7fVxufSIsIl9vIjp7fX0=)

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
Merge branch 'dev' into fix-date-picker

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

default-time property not working when selecting dates multiple times

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

default-time property not working when selecting dates multiple times

warmthsea reviewed (approved):
LGTM, Thanks!

warmthsea reviewed (commented):
- Rel: https://github.com/element-plus/element-plus/issues/3746

If change it manually, the default value will not take effect.

![screenshots](https://github.com/user-attachments/assets/96c1ccb1-aa7b-443a-ae1d-790fbcedbc6a)

SpanManX committed:
Merge branch 'dev' into fix-date-picker

SpanManX committed:
Merge branch 'dev' into fix-date-picker

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

1. the issue has been fixed. Now, even if changed manually, the default value will still take effect.
2. fix test case error validation.

SpanManX commented:
> If change it manually, the default value will not take effect.

1. the issue has been fixed. Now, even if changed manually, the default value will still take effect.
2. fix test case error validation.

warmthsea commented:
Manual changes still have problems.

![screenshots](https://github.com/user-attachments/assets/3394741c-57fa-4993-b0a8-3b45b70d265d)

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

Fix the issue where the value-format causes the default-time attribute to be ineffective.

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

Fix the issue where the value-format causes the default-time attribute to be ineffective.

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

Fix the issue where the value-format causes the default-time attribute to be ineffective.

SpanManX commented:
> Manual changes still have problems.
> 
> ![screenshots](https://private-user-images.githubusercontent.com/45450994/426917296-3394741c-57fa-4993-b0a8-3b45b70d265d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDMxMzcsIm5iZiI6MTc0MzAwMjgzNywicGF0aCI6Ii80NTQ1MDk5NC80MjY5MTcyOTYtMzM5NDc0MWMtNTdmYS00OTkzLWIwYTgtM2I0NWI3MGQyNjVkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE1MjcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMzQzZTRkMmFmN2VjMDZhNjExYzFlMzNiNjhjZWIwYTZjZDkyOWU3ZDVhNWMxMWNkY2Y1ZDdmNzUwZjM5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._fyf90C6IjbzB7J4UylOQpzIOa9kQmfd3NqP-NT-9fs) [ ![screenshots](https://private-user-images.githubusercontent.com/45450994/426917296-3394741c-57fa-4993-b0a8-3b45b70d265d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDMxMzcsIm5iZiI6MTc0MzAwMjgzNywicGF0aCI6Ii80NTQ1MDk5NC80MjY5MTcyOTYtMzM5NDc0MWMtNTdmYS00OTkzLWIwYTgtM2I0NWI3MGQyNjVkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE1MjcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMzQzZTRkMmFmN2VjMDZhNjExYzFlMzNiNjhjZWIwYTZjZDkyOWU3ZDVhNWMxMWNkY2Y1ZDdmNzUwZjM5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._fyf90C6IjbzB7J4UylOQpzIOa9kQmfd3NqP-NT-9fs) ](https://private-user-images.githubusercontent.com/45450994/426917296-3394741c-57fa-4993-b0a8-3b45b70d265d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDMxMzcsIm5iZiI6MTc0MzAwMjgzNywicGF0aCI6Ii80NTQ1MDk5NC80MjY5MTcyOTYtMzM5NDc0MWMtNTdmYS00OTkzLWIwYTgtM2I0NWI3MGQyNjVkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE1MjcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMzQzZTRkMmFmN2VjMDZhNjExYzFlMzNiNjhjZWIwYTZjZDkyOWU3ZDVhNWMxMWNkY2Y1ZDdmNzUwZjM5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._fyf90C6IjbzB7J4UylOQpzIOa9kQmfd3NqP-NT-9fs) [ ](https://private-user-images.githubusercontent.com/45450994/426917296-3394741c-57fa-4993-b0a8-3b45b70d265d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDMxMzcsIm5iZiI6MTc0MzAwMjgzNywicGF0aCI6Ii80NTQ1MDk5NC80MjY5MTcyOTYtMzM5NDc0MWMtNTdmYS00OTkzLWIwYTgtM2I0NWI3MGQyNjVkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE1MjcxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWNkMzQzZTRkMmFmN2VjMDZhNjExYzFlMzNiNjhjZWIwYTZjZDkyOWU3ZDVhNWMxMWNkY2Y1ZDdmNzUwZjM5NTMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0._fyf90C6IjbzB7J4UylOQpzIOa9kQmfd3NqP-NT-9fs)

Fix the issue where the value-format causes the default-time attribute to be ineffective.
![动画](https://github.com/user-attachments/assets/eb68ae6f-11f0-4810-ad93-e8a22c9dc428)

SpanManX commented:
> Manual changes still have problems.
> 
> ![screenshots](https://private-user-images.githubusercontent.com/45450994/426917296-3394741c-57fa-4993-b0a8-3b45b70d265d.gif?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDMwMDUxNjAsIm5iZiI6MTc0MzAwNDg2MCwicGF0aCI6Ii80NTQ1MDk5NC80MjY5MTcyOTYtMzM5NDc0MWMtNTdmYS00OTkzLWIwYTgtM2I0NWI3MGQyNjVkLmdpZj9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTAzMjYlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUwMzI2VDE2MDEwMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTY4MTUxYTdkNDRkZDQzMTc4MjExMGJkNDU2NzE1ZGZkNWQwNTRjNTI1MzhhNmIyMjZkYzRkN2FmNWJmMDIxOGMmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.Z8YuIqhxtleMJJAIxisANLjtAH0SpaLewdRkoKqvBds)

**Fix the issue where the value-format causes the default-time attribute to be ineffective.**
![动画](https://github.com/user-attachments/assets/7b248b00-e29b-45eb-9e47-e64bc16ab9ae)

btea commented on packages/components/time-picker/src/common/picker.vue:
```diff
@@ -202,6 +203,10 @@ import type {
+import type { InputInstance } from '@element-plus/components/input'
+import type { ComponentPublicInstance, Ref } from 'vue'
+import type { Dayjs } from 'dayjs'
+import type { Options } from '@popperjs/core'
```
The order of these import files does not appear to need to be changed.

btea commented on packages/components/time-picker/src/common/picker.vue:
```diff
@@ -313,7 +318,9 @@ const emitInput = (input: SingleOrRange<DateModelType> | null) => {
-    emit(UPDATE_MODEL_EVENT, input ? formatted : input, lang.value)
+
+    const date = input ? formatted : input
+    emit(UPDATE_MODEL_EVENT, date, lang.value)
```
There is no need to modify this variable. There is no place to reuse it, so there is no need.

btea commented on packages/components/time-picker/src/common/picker.vue:
```diff
@@ -540,7 +547,34 @@ const handleChange = () => {
+        let date = dayOrDaysToDate(value)
+
+        if (props.defaultTime) {
+          if (Array.isArray(date) && date.length) {
```
```suggestion
          if (isArray(date) && date.length) {
```

SpanManX committed:
Merge branch 'dev' into fix-date-picker

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
Merge branch 'dev' into fix-date-picker

btea commented on packages/components/date-picker/src/date-picker-com/panel-date-pick.vue:
```diff
@@ -292,12 +292,7 @@ const checkDateWithinRange = (date: ConfigType) => {
-    !isChangeToNow.value &&
-    !isShortcut
-  ) {
+  if (defaultTime && !isChangeToNow.value && !isShortcut) {
```
Why do we need to modify the original judgment logic here?

SpanManX replied:
在 `default-time` 有值的情况下，`!visibleTime.value` 只有一次为 `true`，而其他判断条件则是一直为 `true`。而且在 `type="date"` 的情况下 `emitDayjs` 接收到的参数是 `xxxx-xx-xx 00:00:00`。删除 `!visibleTime.value`  可以保证 `defaultTimeD.value` 能得到执行，而且也不影响 `type="datetime"` 的情况

![image](https://github.com/user-attachments/assets/da509a20-8454-4f5c-a11c-683da63cf58f)

SpanManX replied:
不修改原有条件判断的情况下，另一种解决方法是，新增 `props.type === 'date'` 判断  
`if (
    (!visibleTime.value || props.type === 'date') &&
    defaultTime &&
    !isChangeToNow.value &&
    !isShortcut
  )`

btea commented on packages/components/time-picker/src/common/picker.vue:
```diff
@@ -185,8 +185,14 @@ import {
+  setDateWithDefaultTime,
+  valueEquals,
+} from '../utils'
+import { type DateOrDates, timePickerDefaultProps } from './props'
```
Type imports go below variable imports.

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
fix(components): [date-picker] Fix default-time bug

fix(components): [date-picker] Fix default-time bug

SpanManX committed:
Merge branch 'dev' into fix-date-picker

warmthsea commented on packages/components/time-picker/src/common/picker.vue:
```diff
@@ -540,7 +547,33 @@ const handleChange = () => {
+          if (isArray(date) && date.length) {
+            let arr: DateOrDates | Array<DateOrDates>
+
+            if (isArray(props.defaultTime) && props.defaultTime.length) {
+              arr = date.map((item, i) =>
+                setDateWithDefaultTime(
+                  item as Date,
+                  (props.defaultTime as Date[])[i]
+                )
+              )
```
这段代码似乎不会生效.

<hr />

另外在 daterange 下设置了 default-time 手动更改还是会丢失，这这行代码相关，看起来要在这里增加判断

https://github.com/spanmanx/element-plus/blob/fix-date-picker/packages/components/time-picker/src/common/picker.vue#L714

![screenshots](https://github.com/user-attachments/assets/7666302c-ae34-4709-a1fb-0136171fe4c9)

[play demo](https://element-plus.run/#eyJBcHAudnVlIjoiPHNjcmlwdCBzZXR1cCBsYW5nPVwidHNcIj5cbmltcG9ydCB7IEVsZW1lbnRQbHVzIH0gZnJvbSAnQGVsZW1lbnQtcGx1cy9pY29ucy12dWUnXG5pbXBvcnQgeyB2ZXJzaW9uIGFzIGVwVmVyc2lvbiB9IGZyb20gJ2VsZW1lbnQtcGx1cydcbmltcG9ydCB7IHJlZiwgdmVyc2lvbiBhcyB2dWVWZXJzaW9uIH0gZnJvbSAndnVlJ1xuXG5jb25zdCBkZWZhdWx0VGltZTEgPSBuZXcgRGF0ZSgyMDAwLCAxLCAxLCAxMiwgMCwgMCkgLy8gJzEyOjAwOjAwJ1xuY29uc3QgdmFsdWUxID0gcmVmKCcnKVxuY29uc3QgdmFsdWUyID0gcmVmKCcnKVxuY29uc3QgdmFsdWUzID0gcmVmKCcnKVxuPC9zY3JpcHQ+XG5cbjx0ZW1wbGF0ZT5cbiAgPGgxPldyb25nPC9oMT5cbiAgPHA+e3sgdmFsdWUxIH19PC9wPlxuICA8ZWwtZGF0ZS1waWNrZXJcbiAgICB2LW1vZGVsPVwidmFsdWUxXCJcbiAgICB0eXBlPVwiZGF0ZVwiXG4gICAgcGxhY2Vob2xkZXI9XCJQaWNrIGEgZGF5XCJcbiAgICA6ZGVmYXVsdC10aW1lPVwibmV3IERhdGUoMjAwMCwgMSwgMSwgMjMsIDU5LCA1OSlcIlxuICAgIHZhbHVlLWZvcm1hdD1cIllZWVktTU0tREQgSEg6bW06c3NcIlxuICAvPlxuICA8ZWwtZGF0ZS1waWNrZXJcbiAgICB2LW1vZGVsPVwidmFsdWUzXCJcbiAgICB0eXBlPVwiZGF0ZXJhbmdlXCJcbiAgICByYW5nZS1zZXBhcmF0b3I9XCJUb1wiXG4gICAgc3RhcnQtcGxhY2Vob2xkZXI9XCJTdGFydCBkYXRlXCJcbiAgICBlbmQtcGxhY2Vob2xkZXI9XCJFbmQgZGF0ZVwiXG4gICAgOmRlZmF1bHQtdGltZT1cIltkZWZhdWx0VGltZTEsIGRlZmF1bHRUaW1lMV1cIlxuICAgIHZhbHVlLWZvcm1hdD1cIllZWVktTU0tREQgSEg6bW06c3NcIlxuICAvPlxuICB7eyB2YWx1ZTMgfX1cblxuICA8aDE+UmlnaHQ8L2gxPlxuICA8cD57eyB2YWx1ZTIgfX08L3A+XG4gIDxlbC1kYXRlLXBpY2tlclxuICAgIHYtbW9kZWw9XCJ2YWx1ZTJcIlxuICAgIHR5cGU9XCJkYXRldGltZXJhbmdlXCJcbiAgICBzdGFydC1wbGFjZWhvbGRlcj1cIlN0YXJ0IERhdGVcIlxuICAgIGVuZC1wbGFjZWhvbGRlcj1cIkVuZCBEYXRlXCJcbiAgICA6ZGVmYXVsdC10aW1lPVwiZGVmYXVsdFRpbWUxXCJcbiAgICB2YWx1ZS1mb3JtYXQ9XCJZWVlZLU1NLUREIEhIOm1tOnNzXCJcbiAgLz5cbiAgPHA+XG4gICAgPGVsLWljb24gY29sb3I9XCJ2YXIoLS1lbC1jb2xvci1wcmltYXJ5KVwiPjxFbGVtZW50UGx1cyAvPjwvZWwtaWNvbj5cbiAgICBFbGVtZW50IFBsdXMge3sgZXBWZXJzaW9uIH19ICsgVnVlIHt7IHZ1ZVZlcnNpb24gfX1cbiAgPC9wPlxuPC90ZW1wbGF0ZT5cbiIsImVsZW1lbnQtcGx1cy5qcyI6ImltcG9ydCBFbGVtZW50UGx1cyBmcm9tICdlbGVtZW50LXBsdXMnXG5pbXBvcnQgeyBnZXRDdXJyZW50SW5zdGFuY2UgfSBmcm9tICd2dWUnXG5cbmxldCBpbnN0YWxsZWQgPSBmYWxzZVxuYXdhaXQgbG9hZFN0eWxlKClcblxuZXhwb3J0IGZ1bmN0aW9uIHNldHVwRWxlbWVudFBsdXMoKSB7XG4gIGlmIChpbnN0YWxsZWQpIHJldHVyblxuICBjb25zdCBpbnN0YW5jZSA9IGdldEN1cnJlbnRJbnN0YW5jZSgpXG4gIGluc3RhbmNlLmFwcENvbnRleHQuYXBwLnVzZShFbGVtZW50UGx1cylcbiAgaW5zdGFsbGVkID0gdHJ1ZVxufVxuXG5leHBvcnQgZnVuY3Rpb24gbG9hZFN0eWxlKCkge1xuICBjb25zdCBzdHlsZXMgPSBbJ2h0dHBzOi8vcHJldmlldy0yMDE4OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzJywgJ2h0dHBzOi8vcHJldmlldy0yMDE4OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL3RoZW1lLWNoYWxrL2RhcmsvY3NzLXZhcnMuY3NzJ10ubWFwKChzdHlsZSkgPT4ge1xuICAgIHJldHVybiBuZXcgUHJvbWlzZSgocmVzb2x2ZSwgcmVqZWN0KSA9PiB7XG4gICAgICBjb25zdCBsaW5rID0gZG9jdW1lbnQuY3JlYXRlRWxlbWVudCgnbGluaycpXG4gICAgICBsaW5rLnJlbCA9ICdzdHlsZXNoZWV0J1xuICAgICAgbGluay5ocmVmID0gc3R5bGVcbiAgICAgIGxpbmsuYWRkRXZlbnRMaXN0ZW5lcignbG9hZCcsIHJlc29sdmUpXG4gICAgICBsaW5rLmFkZEV2ZW50TGlzdGVuZXIoJ2Vycm9yJywgcmVqZWN0KVxuICAgICAgZG9jdW1lbnQuYm9keS5hcHBlbmQobGluaylcbiAgICB9KVxuICB9KVxuICByZXR1cm4gUHJvbWlzZS5hbGxTZXR0bGVkKHN0eWxlcylcbn1cbiIsInRzY29uZmlnLmpzb24iOiJ7XG4gIFwiY29tcGlsZXJPcHRpb25zXCI6IHtcbiAgICBcInRhcmdldFwiOiBcIkVTTmV4dFwiLFxuICAgIFwianN4XCI6IFwicHJlc2VydmVcIixcbiAgICBcIm1vZHVsZVwiOiBcIkVTTmV4dFwiLFxuICAgIFwibW9kdWxlUmVzb2x1dGlvblwiOiBcIkJ1bmRsZXJcIixcbiAgICBcInR5cGVzXCI6IFtcImVsZW1lbnQtcGx1cy9nbG9iYWwuZC50c1wiXSxcbiAgICBcImFsbG93SW1wb3J0aW5nVHNFeHRlbnNpb25zXCI6IHRydWUsXG4gICAgXCJhbGxvd0pzXCI6IHRydWUsXG4gICAgXCJjaGVja0pzXCI6IHRydWVcbiAgfSxcbiAgXCJ2dWVDb21waWxlck9wdGlvbnNcIjoge1xuICAgIFwidGFyZ2V0XCI6IDMuM1xuICB9XG59XG4iLCJQbGF5Z3JvdW5kTWFpbi52dWUiOiI8c2NyaXB0IHNldHVwPlxuaW1wb3J0IEFwcCBmcm9tICcuL0FwcC52dWUnXG5pbXBvcnQgeyBzZXR1cEVsZW1lbnRQbHVzIH0gZnJvbSAnLi9lbGVtZW50LXBsdXMuanMnXG5zZXR1cEVsZW1lbnRQbHVzKClcbjwvc2NyaXB0PlxuXG48dGVtcGxhdGU+XG4gIDxBcHAgLz5cbjwvdGVtcGxhdGU+XG4iLCJpbXBvcnQtbWFwLmpzb24iOiJ7XG4gIFwiaW1wb3J0c1wiOiB7XG4gICAgXCJ2dWVcIjogXCJodHRwczovL2Zhc3RseS5qc2RlbGl2ci5uZXQvbnBtL0B2dWUvcnVudGltZS1kb21AbGF0ZXN0L2Rpc3QvcnVudGltZS1kb20uZXNtLWJyb3dzZXIuanNcIixcbiAgICBcIkB2dWUvc2hhcmVkXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AdnVlL3NoYXJlZEBsYXRlc3QvZGlzdC9zaGFyZWQuZXNtLWJ1bmRsZXIuanNcIixcbiAgICBcImVsZW1lbnQtcGx1c1wiOiBcImh0dHBzOi8vcHJldmlldy0yMDE4OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguZnVsbC5taW4ubWpzXCIsXG4gICAgXCJlbGVtZW50LXBsdXMvXCI6IFwidW5zdXBwb3J0ZWRcIixcbiAgICBcIkBlbGVtZW50LXBsdXMvaWNvbnMtdnVlXCI6IFwiaHR0cHM6Ly9mYXN0bHkuanNkZWxpdnIubmV0L25wbS9AZWxlbWVudC1wbHVzL2ljb25zLXZ1ZUAyL2Rpc3QvaW5kZXgubWluLmpzXCJcbiAgfSxcbiAgXCJzY29wZXNcIjoge31cbn0iLCJfbyI6eyJzaG93SGlkZGVuIjp0cnVlLCJzdHlsZVNvdXJjZSI6Imh0dHBzOi8vcHJldmlldy0yMDE4OC1lbGVtZW50LXBsdXMuc3VyZ2Uuc2gvYnVuZGxlL2Rpc3QvaW5kZXguY3NzIn19)

SpanManX replied:
```
         if (isArray(props.defaultTime) && props.defaultTime.length) {
              arr = date.map((item, i) =>
                setDateWithDefaultTime(
                  item as Date,
                  (props.defaultTime as Date[])[i]
                )
              )          
```
el-date-picker 以下情况会触发 `if (isArray(props.defaultTime) && props.defaultTime.length)`

`<el-date-picker v-model="value2" type="datetimerange" start-placeholder="Start Date" end-placeholder="End Date" :default-time="defaultTime2"  />`
`  
const defaultTime2: [Date, Date] = [
  new Date(2000, 1, 1, 12, 0, 0),
  new Date(2000, 2, 1, 8, 0, 0),
]`

![screenshot-20250430-154353](https://github.com/user-attachments/assets/b17d31f8-a1d2-4985-8ceb-2c236f4d9f37)

---

`在 daterange 下设置了 default-time 手动更改还是会丢失`   这个问题得等五一过完我才能修改了😁

SpanManX committed:
Merge branch 'dev' into fix-date-picker

SpanManX committed:
fix(components): [date-picker] Fix default-time bug and test file

SpanManX committed:
Merge branch 'dev' into fix-date-picker
